File: Easy/2894. Divisible and Non-divisible Sums Difference/Go/main.go
func differenceOfSums(n int, m int) int {
    // Total sum from 1 to n
    totalSum := n * (n + 1) / 2

    // Number of multiples of m in [1, n]
    count := n / m

    // Sum of all numbers divisible by m
    divisibleSum := m * count * (count + 1) / 2

    // Return the difference: totalSum - 2 * divisibleSum
    return totalSum - 2 * divisibleSum
}

File: Easy/2894. Divisible and Non-divisible Sums Difference/Rust/main.rs
impl Solution {
    pub fn difference_of_sums(n: i32, m: i32) -> i32 {
        // Total sum from 1 to n
        let total_sum = n * (n + 1) / 2;

        // Count of numbers divisible by m
        let count = n / m;

        // Sum of all numbers divisible by m
        let divisible_sum = m * count * (count + 1) / 2;

        // Return the final difference
        total_sum - 2 * divisible_sum
    }
}

